nemo/collections/nlp/models/question_answering/qa_bert_model.py

```diff
@@ -0,0 +1,892 @@
+from nemo.collections.nlp.modules.common import TokenClassifier
+from nemo.collections.nlp.parts.utils_funcs import tensor2list
+from nemo.core.classes.common import PretrainedModelInfo, typecheck
+from nemo.utils import logging
+
+
+class BERTQAModel(NLPModel):
+    """ BERT model with a QA (token classification) head """
+
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+
+        super().__init__(cfg=cfg, trainer=trainer)
+
+        self.classifier = TokenClassifier(
+            hidden_size=self.hidden_size,
+            num_classes=cfg.token_classifier.num_classes,
+            num_layers=cfg.token_classifier.num_layers,
+            activation=cfg.token_classifier.activation,
+            log_softmax=cfg.token_classifier.log_softmax,
+            dropout=cfg.token_classifier.dropout,
+            use_transformer_init=cfg.token_classifier.use_transformer_init,
+        )
+
+        self.loss = SpanningLoss()
+
+    def setup_training_data(self, train_data_config: Optional[DictConfig]):
+        if not train_data_config or not train_data_config.file:
+            logging.info(
+                f"Dataloader config or file_path for the train is missing, so no data loader for test is created!"
+            )
+            self._test_dl = None
+            return
+
+        self._train_dl = self._setup_dataloader_from_config(cfg=train_data_config, mode=TRAINING_MODE)
+
+    def setup_validation_data(self, val_data_config: Optional[DictConfig]):
+        if not val_data_config or not val_data_config.file:
+            logging.info(
+                f"Dataloader config or file_path for the validation is missing, so no data loader for test is created!"
+            )
+            self._test_dl = None
+            return
+
+        self._validation_dl = self._setup_dataloader_from_config(cfg=val_data_config, mode=EVALUATION_MODE)
+
+    def setup_test_data(self, test_data_config: Optional[DictConfig]):
+        if not test_data_config or test_data_config.file is None:
+            logging.info(
+                f"Dataloader config or file_path for the test is missing, so no data loader for test is created!"
+            )
+            self._test_dl = None
+            return
+
+        self._test_dl = self._setup_dataloader_from_config(cfg=test_data_config, mode=EVALUATION_MODE)
+
+    def training_step(self, batch, batch_idx):
+        input_ids, input_type_ids, input_mask, unique_ids, start_positions, end_positions = batch
+        logits = self.forward(input_ids=input_ids, token_type_ids=input_type_ids, attention_mask=input_mask)
+        loss, _, _ = self.loss(logits=logits, start_positions=start_positions, end_positions=end_positions)
+        lr = self._optimizer.param_groups[0]['lr']
+        self.log('train_loss', loss)
+        self.log('lr', lr, prog_bar=True)
+
+        return {'loss': loss, 'lr': lr}
+
+    def validation_step(self, batch, batch_idx):
+        if self.trainer.testing:
```

Comment:
nit: in line if else can work here `prefix = 'test' if self.trainer.testing else 'val'`